stdlib/streams/src/main/ballerina/streams/snapshot.bal

```diff
@@ -0,0 +1,291 @@
+function purgeOldSnapshotFiles(string persistancePath) {
+    int revisionsToKeep = 5;
+    string[] files = getSnapshotFiles(persistancePath);
+    if (files.length() > revisionsToKeep) {
+        int i = 0;
+        while (i < (files.length() - revisionsToKeep)) {
+            string f = files[i];
+            internal:Path p = new(f);
+            if (p.exists()) {
+                error? e = p.delete();
+                if (e is error) {
+                    io:println("couldn't delete snapshot ", p.getPathValue(), ". ", e.detail());
+                }
+            }
+            i += 1;
+        }
+    }
+}
+
+// Function to restore a persisted snapshotable states from a file.
+function restoreStates() {
+    string? lastSnapshotFile = getLatestSnapshotFile(persistanceDirectory);
+    if (lastSnapshotFile is string) {
+        string stateStr = readFile(lastSnapshotFile, "UTF8");
+        var retrievedState = trap deserialize(stateStr);
+        if (retrievedState is map<map<any>>) {
+            streamsPersistanceState = retrievedState;
+            foreach var (k, v) in streamsPersistanceState {
+                Snapshotable? s = snapshotables[k];
+                if (s is Snapshotable) {
+                    s.restoreState(v);
+                }
+            }
+        }
+    }
+}
+
+// Function to iterate through all the snapshotables, and persist their states into a file.
+function persistStatesAndPurgeOldSnapshots() returns error? {
+    foreach var (k, v) in snapshotables {
+        streamsPersistanceState[k] = v.saveState();
+    }
+    error? e = writeStateToFile(persistanceDirectory);
+    purgeOldSnapshotFiles(persistanceDirectory);
+    return e;
+}
+
+// Function to initilize a Timer task and start periodic snapshotting.
+function startPersisting() {
+    (function () returns error?) onTriggerFunction = persistStatesAndPurgeOldSnapshots;
+    function (error) onErrorFunction = persitanceError;
+    persistTimer = new task:Timer(onTriggerFunction, onErrorFunction, persistanceIntervalInMillis,
+        delay = persistanceIntervalInMillis);
+    persistTimer.start();
+}
+
+// Function to restore state of a given object.
+public function restoreState(string k, any o) {
+    var state = streamsPersistanceState[k];
+    if (state is map<any> && o is Snapshotable) {
+        o.restoreState(state);
+    }
+}
+
+// Function to clear an existing state.
+public function removeState(string k) returns boolean {
+    boolean snapshotableRemoved = snapshotables.remove(k);
+    boolean stateRemoved = streamsPersistanceState.remove(k);
+    return snapshotableRemoved && stateRemoved;
+}
+
+// Function to register Snapshotables.
+public function registerSnapshotable(string key, any ref) {
+    if (ref is Snapshotable) {
+        snapshotables[key] = ref;
+    }
+}
+
+// Function to initialize snapshotting.
+public function initPersistence() {
+    boolean enabled = config:getAsBoolean("b7a.streaming.persistence.enabled", default = false);
```

Comment:
Since `false` is the default we can just say 
```suggestion
    boolean enabled = config:getAsBoolean("b7a.streaming.persistence.enabled");
```
right?

Reply:
fixed in 4858833